### DISCUSSION TITLE
external command used inside each {} loop makes it return a list of closures

### DISCUSSION THREAD

**Discussion by foodornt** [nushell/nushell#8106] [Q&A]
I'm trying to get a list of media files along with their duration
```
ls  | each {|e| {name: $e.name, duration: ffprobe -v error -select_streams v:0 -show_entries stream=duration -of default=noprint_wrappers=1:nokey=1 $e.name }}
```

result:
```
╭─────┬──────────────╮
│   0 │ <Closure 92> │
│   1 │ <Closure 92> │
│   2 │ <Closure 92> │
╰─────┴──────────────╯
```

How do I invoke them?

**Comment by fdncred** [comment]
I think nushell is confused because you're trying to make a record in each loop but your syntax isn't correct. To make a record you need { key: value }. If you have spaces in your value you need to wrap it in quotes. If you have variables in your value, you need to use string interpolation. If you want to execute what you're trying to put in duration, it probably needs to be a string interpolated string beginning with ^.

**Reply by foodornt** [reply]
I needed to put the duration command between parenthesis, it was erroring out some files
now it works like a charm
